src/libcore/iter/traits/iterator.rs

```diff
@@ -1696,6 +1696,46 @@ pub trait Iterator {
         Try::from_ok(accum)
     }
 
+    /// The same as [`try_fold()`](#method.try_fold), but uses the first element
+    /// in the iterator as the initial value, folding every subsequent element
+    /// into it. If the iterator is empty, return None; otherwise, return the
+    /// result of the fold.
+    ///
+    /// # Examples
+    ///
+    /// Find the maximum value:
+    ///
+    /// ```
+    /// use std::cmp::Ordering;
+    /// fn find_max<I>(iter: I) -> Option<I::Item>
+    ///     where I: Iterator,
+    ///           I::Item: PartialCmp,
+    /// {
+    ///     iter.try_fold_self(|a, b| {
+    ///         a.partial_cmp(b).map(move |cmp| match cmp {
+    ///             Ordering::Greater | Ordering::Equal => a,
+    ///             Ordering::Less => b,
+    ///         })
+    ///     })
+    /// }
+    /// let a = [10, 20, 5, -23, 0];
+    /// let b = [10, 20, -23, std::f64::NAN, 12, 2.5];
+    /// let c = [];
+    ///
+    /// assert_eq!(find_max(a.iter()), Some(Some(20)));
+    /// assert_eq!(find_max(b.iter()), Some(None));
+    /// assert_eq!(find_max(c.iter()), None);
+    /// ```
+    #[inline]
+    #[unstable(feature = "iterator_fold_self", issue = "65222")]
+    fn try_fold_self<F, R>(&mut self, f: F) -> Option<R>
+        where Self: Sized,
+              F: FnMut(Self::Item, Self::Item) -> R,
+              R: Try<Ok = Self::Item>
+    {
+        self.next().map(#[inline] move |first| self.try_fold(first, f))
```

Comment:
None of the inline hints are particularly necessary, and I'd want to remove them.

Reply:
Done